app/api/auth/product-page/page.tsx

```diff
@@ -0,0 +1,50 @@
+'use client';
+
+import React, { Component } from 'react';
+
+class AddProduct extends Component {
```

Comment:
the React community moved to functional components from class components.  Let's use functional components instead.

Reply:
Here are some resources

https://www.geeksforgeeks.org/reactjs-functional-components/
https://www.educative.io/blog/react-component-class-vs-functional
https://www.geeksforgeeks.org/differences-between-functional-components-and-class-components-in-react/